tests/trainer/test_data_loading.py

```diff
@@ -74,15 +71,12 @@ def on_test_start(self) -> None:
         def create_dataset(self):
             dataset = IndexedRandomDataset(32, 64)
             if self._mode == 1:
-                # this case will raise an error
-                return FailureCustomDataLoader(32, dataset)
```

Comment:
Reviewer: I removed/renamed the original mode 1 here because what was expected to fail here is now supported.